Feature/trigger patterns (#30)
hashicorp/terraform-docs-common: website/docs/cloud-docs/workspaces/settings/vcs.mdx

## API

You can use the [Update a Workspace endpoint](/cloud-docs/api-docs/workspaces#update-a-workspace) in the Workspaces API to change one or more VCS settings. We also recommend using this endpoint to automate changing VCS connections for many workspaces at once. For example, when you move a VCS server or remove a deprecated API version.

## Version Control Settings

To change a workspace's VCS settings:

1. Go to the workspace and click **Settings > Version Control**. The **Version Control** page appears.
1. Choose the desired settings and click **Update VCS settings**.

You can update the following types of VCS settings for the workspace.

### VCS Connection

You can take one of the following actions:

- To add a new VCS connection, click **Connect to version control**. Select **Version control workflow** and follow the steps to [select a VCS provider and repository](/cloud-docs/workspaces/creating#create-a-workspace).
- To edit an existing VCS connection, click **Change source**. Choose the **Version control workflow** and follow the steps to [select VCS provider and repository](/cloud-docs/workspaces/creating#create-a-workspace).

> - To remove the VCS connection, click **Change source**. Select either the **CLI-driven workflow** or the **API-driven workflow** and click **Update VCS settings**. The workspace is no longer connected to VCS.

Reviewer: ```suggestion
- To remove the VCS connection, click **Change source**. Select either the **CLI-driven workflow** or the **API-driven workflow**, and click **Update VCS settings**. The workspace is no longer connected to VCS.
```

Author: faa3ccb85b5f3d3f14d78be665798f0214f53c93

[permissions-citation]: #intentionally-unused---keep-for-maintainers

### Terraform Working Directory

> Specify the directory where Terraform will execute runs. This defaults to the root directory in your repository, but you may want to specify a root directory if you have directories for multiple different Terraform configurations within the same repository. For example, if you had one `staging` directory and one `production` directory.

Reviewer: ```suggestion
Specify the directory where Terraform will execute runs. This defaults to the root directory in your repository, but you may want to specify another directory if you have directories for multiple different Terraform configurations within the same repository. For example, if you had one `staging` directory and one `production` directory.
```

Author: 9af186f041ef009b763ca0c7f5156b3ed7533859

  !> **Warning:** If you choose auto apply, make sure that no one can change your infrstructure outside of your automated build pipeline. This reduces the risk of congfiguration drift and unexpected changes.

- **Manual apply:** Terraform will ask for approval before applying changes from a successful plan. A push to the default branch of your repository will trigger a plan, and then Terraform will wait for confirmation.

### Automatic Run Triggering

Terraform Cloud uses your VCS provider's API to retrieve the changed files in your repository. You can choose one of the following options to specify which changes trigger Terraform runs.

#### Always trigger runs

This option instructs Terraform to begin a run when changes are pushed to any file within the repository. This can be useful for repositories that do not have multiple configurations but require a working directory for some other reason. However, we do not recommend this approach for true monorepos, as it queues unnecessary runs and slows down your ability to provision infrastructure.

#### Only trigger runs when files in specified paths change

> This option instructs Terraform to begin new runs only for changes that affected specified files and directories. This behavior also applies to [speculative plans](/cloud-docs/run/#speculative-plans) on pull requests.

Reviewer: ```suggestion
This option instructs Terraform to begin new runs only for changes that affect specified files and directories. This behavior also applies to [speculative plans](/cloud-docs/run/#speculative-plans) on pull requests.
```

Author: bcb46142c632e13f077a8be450fae75d6d09992a

You can use trigger patterns and trigger prefixes in the **Add path** field to specify groups of files and directories.

> - **Trigger Patterns:** (Recommended) Use glob patterns to specify the files that should trigger a new run. For example, `/submodule/**/*.tf`, specifies all files with the `.tf` extension that are nested below the `submodule` directory. You can also use more complex patterns like `/**/networking/**/*`, which specifies all files which have a `networking` folder on their file path (e.g., `/submodule/service-1/networking/private/main.tf`). Refer to [Glob Patterns for Automatic Run Triggering](#glob-patterns-for-automatic-run-triggering) for details.

Reviewer: ```suggestion
- **Trigger Patterns:** (Recommended) Use glob patterns to specify the files that should trigger a new run. For example, `/submodule/**/*.tf`, specifies all files with the `.tf` extension that are nested below the `submodule` directory. You can also use more complex patterns like `/**/networking/**/*`, which specifies all files that have a `networking` folder on their file path (e.g., `/submodule/service-1/networking/private/main.tf`). Refer to [Glob Patterns for Automatic Run Triggering](#glob-patterns-for-automatic-run-triggering) for details.
```

> - **Trigger Patterns:** (Recommended) Use glob patterns to specify the files that should trigger a new run. For example, `/submodule/**/*.tf`, specifies all files with the `.tf` extension that are nested below the `submodule` directory. You can also use more complex patterns like `/**/networking/**/*`, which specifies all files in the `networking` folder at this level of the file path (e.g., `/submodule/service-1/networking/private/main.tf`). Refer to [Glob Patterns for Automatic Run Triggering](#glob-patterns-for-automatic-run-triggering) for details.

Author: ```
which specifies all files in the `networking` folder at this level of the file path
```
This sentence is a bit incorrect and I would revert it back to read something like:
```
which specifies all files which have a `networking` folder on their file path
```

@laurapacilio what do you think of it?

Author: 1c7e5b6d6bdf7616b3f82a3846f96dbefe734d45

> - ** Trigger Prefixes:** Terraform Cloud will queue runs for changes in any of the specified trigger directories matching the provided prefixes (including the working directory). For example, if you use a top-level `modules` directory to share Terraform code across multiple configurations, changes to the shared modules are relevant to every workspace that uses that repo. You can add `modules` as a trigger directory for each workspace to make sure they notice any changes to shared code.

Reviewer: ```suggestion
- **Trigger Prefixes:** Terraform Cloud will queue runs for changes in any of the specified trigger directories matching the provided prefixes (including the working directory). For example, if you use a top-level `modules` directory to share Terraform code across multiple configurations, changes to the shared modules are relevant to every workspace that uses that repository. You can add `modules` as a trigger directory for each workspace to track changes to shared code.
```

Author: a5c065b22c5632c104cabbdc8d04a84e9b4413f9

Author: a5c065b22c5632c104cabbdc8d04a84e9b4413f9

-> **Note:** Terraform Cloud triggers runs on all attached workspaces if it does not receive a list of changed files or if that list is too large to process. When this happens, Terraform Cloud may show several runs with completed plans that do not result in infrastructure changes.

### VCS Branch

This designates which branch of the repository Terraform Cloud should use in the workspace. If left blank, Terraform Cloud uses the repository's default branch.

### Automatic Speculative Plans

Whether to perform [speculative plans on pull requests](/cloud-docs/run/ui#speculative-plans-on-pull-requests) to the connected repository, to assist in reviewing proposed changes. Automatic speculative plans are enabled by default, but you can disable them for any workspace.

### Include Submodules on Clone

Select **Include submodules on clone** to recursively clone all of the repository's Git submodules when Terraform Cloud feteches a configuration.

> -> **Note:** The [SSH key for cloning Git submodules](/cloud-docs/vcs/#ssh-keys) is set in the VCS provider settings for the organization, and is not related to the workspace's SSH key for Terraform modules.

Author: 88db13eea76efb907c1c0090f8f2890b398a3321

## Glob Patterns for Automatic Run Triggering

> We support `glob` patterns to describe a set of triggers if `trigger patterns` option is selected.

Reviewer: ```suggestion
We support `glob` patterns to describe a set of triggers for automatic run triggering. Refer to [trigger patterns](#only-trigger-runs-when-files-in-specified-paths-change) for details.
```

Author: 5ea81af788e02f6e212d1c75ee8d2f0d504e4271

> We support `glob` patterns to describe a set of triggers for automatic run triggering. Refer to [trigger patterns](#only-trigger-runs-when-files-in-specified-paths-change) for details.

Reviewer: ```suggestion
We support `glob` patterns to describe a set of triggers for automatic runs. Refer to [trigger patterns](#only-trigger-runs-when-files-in-specified-paths-change) for details.
```

Author: b5f16fd0a8ba12e66b9b47150211a60543378555

Supported wildcards:
`*`  Matches zero or more characters.
`?`  Matches one or more characters.
`**` Matches directories recursively.

> Examples:

Reviewer: ```suggestion
The following examples demonstrate how to use the supported wildcards:
```

> /**/* matches every file in every directory

Reviewer: can these be code formatting? and should they be bullets?

Author: 5ea81af788e02f6e212d1c75ee8d2f0d504e4271
